### PR TITLE
Rename build files to build.gradle

### LINKED ISSUE #3408: Rename build files to build.gradle
I'd like to propose renaming our build files to the standard Gradle default. After working on the repo and probably the most on Gradle files, I've found no real benefit in navigating among build scripts (IDE brings up the folder names too when jumping around) by having the names but have some trouble with copy-pasting files. I believe last year I checked `github.com/gradle/gradle` and they did actually use the non-default file names in that repo, but now they seem to have changed

https://github.com/gradle/gradle/tree/master/subprojects

This is probably for consistency with the general Gradle ecosystem and I feel following the pattern of gradle/gradle is a good one for us. This logic is a bit massive for what we get of it IMO

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/settings.gradle#L319

### PR DESCRIPTION
**Pull request by anuraaga**
```
zmv '(^(examples|smoke-tests))/(**/)*.gradle' '$1/$2build.gradle'
mv smoke-tests/smoke-tests.gradle smoke-tests/build.gradle
```

Fixes #3408

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]
🎉

File: instrumentation/external-annotations/javaagent-unit-tests/src/test/groovy/IncludeTest.groovy [hunk 1]
```diff
@@ -3,11 +3,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import static io.opentelemetry.javaagent.instrumentation.extannotations.TraceAnnotationsInstrumentationModule.DEFAULT_ANNOTATIONS
+import static io.opentelemetry.javaagent.instrumentation.extannotations.ExternalAnnotationInstrumentation.DEFAULT_ANNOTATIONS
```
Comment (anuraaga) [inline instrumentation/external-annotations/javaagent-unit-tests/src/test/groovy/IncludeTest.groovy:6]: Some build cache problem 😮

Comment (anuraaga) [reply instrumentation/external-annotations/javaagent-unit-tests/src/test/groovy/IncludeTest.groovy:6]: Nevermind, the old build file had a typo in its name!

Comment (anuraaga) [reply instrumentation/external-annotations/javaagent-unit-tests/src/test/groovy/IncludeTest.groovy:6]: Independently I had already filed https://github.com/gradle/gradle/issues/17540

**Pull request merged**